beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -917,12 +917,24 @@ impl<AppReqId: ReqId, E: EthSpec> Network<AppReqId, E> {
     /* Eth2 RPC behaviour functions */
 
     /// Send a request to a peer over RPC.
-    pub fn send_request(&mut self, peer_id: PeerId, request_id: AppReqId, request: Request) {
-        self.eth2_rpc_mut().send_request(
-            peer_id,
-            RequestId::Application(request_id),
-            request.into(),
-        )
+    pub fn send_request(
+        &mut self,
+        peer_id: PeerId,
+        request_id: AppReqId,
+        request: Request,
+    ) -> Result<(), (AppReqId, RPCError)> {
+        // Check if the peer is connected before sending an RPC request
+        if self.swarm.is_connected(&peer_id) {
```

Comment:
nit, wdyt of:
```suggestion
        if !self.swarm.is_connected(&peer_id) {
            return Err((request_id, RPCError::Disconnected));
         }
```
I'd argue it's easier to read as we know one line below what happens if the peer is not connected and then we return to the main context of the function instead.